token-swap/program/src/processor.rs

```diff
@@ -34,19 +34,23 @@ const TOKEN_PROGRAM_ID: Pubkey = Pubkey::new_from_array([1u8; 32]);
 pub struct Processor {}
 impl Processor {
     /// Deserializes a spl_token `Account`.
-    pub fn token_account_deserialize(data: &[u8]) -> Result<spl_token::state::Account, Error> {
-        spl_token::state::Account::unpack(data).map_err(|_| Error::ExpectedAccount)
+    pub fn token_account_deserialize(data: &[u8]) -> Result<spl_token::state::Account, SwapError> {
```

Comment:
Now that we're using `unpack` everywhere, maybe rename these methods?
Suggest: `unpack_token_account` and `unpack_mint`

Reply:
Definitely